core/templates/dev/head/pages/exploration_player/AnswerClassificationService.js

```diff
@@ -61,30 +58,32 @@ oppia.factory('AnswerClassificationService', [
           if (rule.type !== RULE_TYPE_CLASSIFIER &&
               interactionRulesService[rule.type](
                 answer, rule.inputs)) {
-            return {
-              outcome: answerGroups[i].outcome,
-              answerGroupIndex: i,
-              ruleIndex: j,
-              classificationCategorization: EXPLICIT_CLASSIFICATION
-            };
+            return AnswerClassificationResult.createNew(
+              answerGroups[i].outcome, i, j, EXPLICIT_CLASSIFICATION);
           }
         }
       }
 
       // If no rule in any answer group returns true, the default 'group' is
       // returned. Throws an error if the default outcome is not defined.
       if (defaultOutcome) {
-        return {
-          outcome: defaultOutcome,
-          answerGroupIndex: answerGroups.length,
-          ruleIndex: 0,
-          classificationCategorization: DEFAULT_OUTCOME_CLASSIFICATION
-        };
+        return AnswerClassificationResult.createNew(
+          defaultOutcome, answerGroups.length, 0, DEFAULT_OUTCOME_CLASSIFICATION
+        );
       } else {
         alertsService.addWarning('Something went wrong with the exploration.');
       }
     };
 
+    var findRuleIndex = function(answerGroup, answerGroupIndex) {
+      for (var i = 0; i < answerGroup.rules.length; i++) {
+        var rule = answerGroup.rules[i];
+        if (rule.type == RULE_TYPE_CLASSIFIER) {
+          return i;
+        }
+      }
+    };
```

Comment:
What do you want to return if it's not found? Null? -1?

Consider adding a short docstring above the function to explain what the function does and what convention you use here.